src/customer/strategies/square-customer-strategy.ts

```diff
@@ -0,0 +1,36 @@
+import {CheckoutStore, InternalCheckoutSelectors} from '../../checkout';
+import {NotImplementedError} from '../../common/error/errors';
+import {RemoteCheckoutActionCreator} from '../../remote-checkout';
+import CustomerCredentials from '../customer-credentials';
+import {CustomerRequestOptions} from '../customer-request-options';
+
+import CustomerStrategy from './customer-strategy';
+
+export default class SquareCustomerStrategy extends CustomerStrategy {
+
+    constructor(
+        store: CheckoutStore,
+        private _remoteCheckoutActionCreator: RemoteCheckoutActionCreator
+    ) {
+        super(store);
+    }
+
+    signIn(credentials: CustomerCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        throw new NotImplementedError(
+            'In order to sign in via Masterpass, the shopper must click on "Masterpass" button.'
+        );
+    }
+
+    signOut(options?: any): Promise<InternalCheckoutSelectors> {
+        const state = this._store.getState();
+        const payment = state.payment.getPaymentId();
+
+        if (!payment) {
+            return Promise.resolve(this._store.getState());
+        }
+
+        return this._store.dispatch(
+            this._remoteCheckoutActionCreator.signOut(payment.providerId, options)
```

Comment:
Why do we have this file? Do we want to offer a sign-in button at the customer step? It doesn't seem to be.

Reply:
Actually, we've created this file to offer the sign out link

Reply:
Oh okay, I thought you can just load the Masterpass button without loading other CC iframes.

Anyway, a question, what happens when you click on the sign out link? Does it log you out of Masterpass, as well as resetting all the pre-populated checkout data (i.e.: email, shipping, billing details etc...)?

Reply:
Affirmative.